packages/devtools_app/lib/src/http/http_request_data.dart

```diff
@@ -361,3 +448,164 @@ class HttpRequestData extends NetworkRequest {
+      if (_request.response != null) ...{
+        'compressionState': _request.response.compressionState,
+        'isRedirect': _request.response.isRedirect,
+        'persistentConnection': _request.response.persistentConnection,
+        'reasonPhrase': _request.response.reasonPhrase,
+        'redirects': _request.response.redirects,
+        'statusCode': _request.response.statusCode,
+      },
+    };
+  }
+
+  @override
+  String get method => _request.method;
+
+  /// True if the HTTP request hasn't completed yet, determined by the lack of
+  /// an end event.
+  @override
+  bool get inProgress => !_request.isResponseComplete;
+
+  /// All instant events logged to the timeline for this HTTP request.
+  @override
+  List<HttpInstantEvent> get instantEvents {
+    if (_instantEvents == null) {
+      _instantEvents = [
+        for (final event in _request.request?.events ?? [])
+          DartIOHttpInstantEvent._(event)
+      ];
+      _recalculateInstantEventTimes();
+    }
+    return _instantEvents;
+  }
+
+  List<HttpInstantEvent> _instantEvents;
+
+  /// A list of all cookies contained within the request headers.
+  @override
+  List<Cookie> get requestCookies =>
+      HttpRequestData._parseCookies(_request.request?.cookies);
+
+  /// The request headers for the HTTP request.
+  @override
+  Map<String, dynamic> get requestHeaders => _request.request?.headers;
+
+  /// A list of all cookies contained within the response headers.
+  @override
+  List<Cookie> get responseCookies =>
+      HttpRequestData._parseCookies(_request.response?.cookies);
+
+  /// The response headers for the HTTP request.
+  @override
+  Map<String, dynamic> get responseHeaders => _request.response?.headers;
+
+  /// Merges the information from another [HttpRequestData] into this instance.
+  @override
+  void merge(HttpRequestData data) {
+    assert(data is DartIOHttpRequestData);
+    final requestData = data as DartIOHttpRequestData;
+    _request = requestData._request;
+  }
+
+  @override
+  DateTime get endTimestamp => DateTime.fromMicrosecondsSinceEpoch(
+        timelineMicrosecondsSinceEpoch(_request?.response?.endTime),
+      );
+
+  @override
+  DateTime get startTimestamp => DateTime.fromMicrosecondsSinceEpoch(
+        timelineMicrosecondsSinceEpoch(_request.startTime),
+      );
+
+  @override
+  String get status => _request.response?.statusCode?.toString();
+
+  @override
+  String get uri => _request.uri.toString();
+
+  @override
+  String get responseBody {
+    if (_request is! HttpProfileRequest) {
+      return null;
+    }
+    final fullRequest = _request as HttpProfileRequest;
+    try {
+      if (!_request.isResponseComplete) return null;
+      if (_responseBody != null) return _responseBody;
+      _responseBody = utf8.decode(fullRequest.responseBody);
+      if (contentType.contains('json')) {
+        print('formatting json');
+        _responseBody = const JsonEncoder.withIndent('  ').convert(json.decode(_responseBody));
```

Comment:
Let's use `FormattedJson.encoder` to make sure we use the same encoder throughout DevTools. Can you also add a TODO that says consider returning a `Map<String, dynamic>` from responseBody so the `FormattedJson` widget can be used directly in the UI?

Reply:
Not sure we'd be able to return `Map<String, dynamic>` in the general case since a simple string response is completely valid.